internal/providers/pluginfw/resources/sharing/resource_acc_test.go

```diff
@@ -0,0 +1,140 @@
+		properties = {
+			purpose = "testing"
+		}
+	}
+
+	resource "databricks_schema" "things" {
+		catalog_name = databricks_catalog.sandbox.id
+		name         = "things{var.STICKY_RANDOM}"
+		comment      = "this database is managed by terraform"
+		properties = {
+			kind = "various"
+		}
+	}
+
+	resource "databricks_table" "mytable" {
+		catalog_name = databricks_catalog.sandbox.id
+		schema_name = databricks_schema.things.name
+		name = "bar"
+		table_type = "MANAGED"
+		data_source_format = "DELTA"
+		
+		column {
+			name      = "id"
+			position  = 0
+			type_name = "INT"
+			type_text = "int"
+			type_json = "{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}"
+		}
+	}
+
+	resource "databricks_table" "mytable_2" {
+		catalog_name = databricks_catalog.sandbox.id
+		schema_name = databricks_schema.things.name
+		name = "bar_2"
+		table_type = "MANAGED"
+		data_source_format = "DELTA"
+		
+		column {
+			name      = "id"
+			position  = 0
+			type_name = "INT"
+			type_text = "int"
+			type_json = "{\"name\":\"id\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}"
+		}
+	}
+`
+
+const preTestTemplateUpdate = `
+	resource "databricks_grants" "some" {
+		catalog = databricks_catalog.sandbox.id
+		grant {
+			principal  = "account users"
+			privileges = ["ALL_PRIVILEGES"]
+		}
+		grant {
+			principal  = "{env.TEST_METASTORE_ADMIN_GROUP_NAME}"
+			privileges = ["ALL_PRIVILEGES"]
+		}
+	}
+`
+
+func TestUcAccCreateShare(t *testing.T) {
+	acceptance.UnityWorkspaceLevel(t, acceptance.Step{
+		Template: preTestTemplate + `		
+		resource "databricks_share_pluginframework" "myshare" {
+			name  = "{var.STICKY_RANDOM}-terraform-delta-share"
+			owner = "account users"
+			object {
+				name = databricks_table.mytable.id
+				comment = "c"
+				data_object_type = "TABLE"
+         	}
+			object {
+				name = databricks_table.mytable_2.id
+				cdf_enabled = false
+				comment = "c"
+				data_object_type = "TABLE"
+			}					
+		}
+
+		resource "databricks_recipient" "db2open" {
+			name = "{var.STICKY_RANDOM}-terraform-db2open-recipient"
+			comment = "made by terraform"
+			authentication_type = "TOKEN"
+			sharing_code = "{var.STICKY_RANDOM}"
+			ip_access_list {
+			// using private ip for acc testing
+			allowed_ip_addresses = ["10.0.0.0/16"]
+			}
+		}
+
+		resource "databricks_grants" "some" {
+			share = databricks_share_pluginframework.myshare.name
+			grant {
+				principal  = databricks_recipient.db2open.name
+				privileges = ["SELECT"]
+			}
+		}			
+		`,
+	})
+}
+
+func shareTemplateWithOwner(comment string, owner string) string {
+	return fmt.Sprintf(`
+		resource "databricks_share_pluginframework" "myshare" {
+			name  = "{var.STICKY_RANDOM}-terraform-delta-share"
+			owner = "%s"
+			object {
+				name = databricks_table.mytable.id
+				comment = "%s"
+				data_object_type = "TABLE"
+				history_data_sharing_status = "DISABLED"
+			}							
+		}`, owner, comment)
+}
+
+func TestUcAccUpdateShare(t *testing.T) {
```

Comment:
Can you please add an integration test which adds an object in middle in the update step? This was a limitation in SDKv2 as ordering used to matter whereas now it should work with plugin framework.

Reply:
I think this would fail currently because object in ShareInfo that we use for schema is a list and it should be set instead. 

````

type ShareInfoExtended struct {
	sharing_tf.ShareInfo
}
````